ironfish/src/network/peers/peerManager.ts

```diff
@@ -178,19 +216,26 @@ export class PeerManager {
       return false
     }
 
-    // If we're trying to connect to the peer, we don't care about limiting the peer's connections to us
-    peer.localRequestedDisconnectUntil = null
-    peer.localRequestedDisconnectReason = null
+    const alternateIdentity = peer.state.identity ?? peer.getWebSocketAddress()
 
-    // Clear out peerRequestedDisconnect if we passed it
-    peer.peerRequestedDisconnectUntil = null
-    peer.peerRequestedDisconnectReason = null
+    const candidate = this.peerCandidateMap.get(alternateIdentity)
```

Comment:
This is a little confusing that we store a mapping in `peerCandidateMap` by both web socket address and peer identity. Would this ever result in duplicates? I guess it makes sense that we'd need to use web socket address because some of these peers are unidentified but if that's the case then couldn't we use web socket address for all of them and have some separate mapping of peerIdentity to web socket address? Just seems a little brittle to map by both

Reply:
This is used for the bootstrap node primarily since it does not have an identity. We will consider refactoring connecting to the bootstrap nodes in the future but leave this as is for now